src/machine-c-api.h

```diff
@@ -82,1894 +83,544 @@ typedef enum { // NOLINT(modernize-use-using)
+/// \param runtime_config Machine runtime configuration as a JSON string.
+/// \param new_machine Receives the pointer to new machine instance.
+/// \returns 0 for success, non zero code for error.
+CM_API int cm_create(const char *config, const char *runtime_config, cm_machine **new_machine);
+
+/// \brief Destroys a machine.
+/// \returns 0 for success, non zero code for error.
+/// \details The machine is deallocated and its pointer cannot be used after this call.
+CM_API int cm_destroy(cm_machine *m);
+
+/// \brief Creates a machine instance from previously serialized directory.
+/// \param dir Directory where previous machine is serialized.
+/// \param runtime_config Machine runtime configuration as a JSON string.
+/// \param new_machine Receives the pointer to new machine instance.
+/// \returns 0 for success, non zero code for error.
+CM_API int cm_load(const char *dir, const char *runtime_config, cm_machine **new_machine);
+
+/// \brief Serializes entire machine state to a directory.
+/// \param m Pointer to a valid machine instance.
 /// \param dir Directory where the machine will be serialized
-/// \param err_msg Receives the error message if function execution fails
-/// or NULL in case of successful function execution. In case of failure error_msg
-/// must be deleted by the function caller using cm_delete_cstring.
-/// err_msg can be NULL, meaning the error message won't be received.
-/// \details The method changes machine because it updates the root hash
-/// \returns 0 for success, non zero code for error
-CM_API int cm_store(cm_machine *m, const char *dir, char **err_msg);
-
-/// \brief Deletes machine instance
-/// \param m Valid pointer to the existing machine instance
-CM_API void cm_delete_machine(cm_machine *m);
+/// \details The method changes machine because it updates the root hash.
+/// \returns 0 for success, non zero code for error.
+CM_API int cm_store(cm_machine *m, const char *dir);
+
+/// \brief Returns the machine initial config.
+/// \param m Pointer to a valid machine instance.
+/// \param config Receives the initial configuration as a JSON string,
+/// remains valid until the machine is destroyed.
+/// \returns 0 for success, non zero code for error.
+/// The string returned by this function must not be changed nor deallocated.
+CM_API int cm_get_initial_config(const cm_machine *m, const char **config);
+
+/// \brief Replaces a memory range.
+/// \param m Pointer to a valid machine instance.
+/// \param start Memory start physical address.
+/// \param length Memory length.
+/// \param shared If true changes by the machine will be shared to the image file.
+/// \param image_filename Image file name to load into the range.
+/// \returns 0 for success, non zero code for error.
+/// \details The machine must contain an existing memory range matching the start and length
+/// specified in new range.
+CM_API int cm_replace_memory_range(cm_machine *m, uint64_t start, uint64_t length, bool shared,
```

Comment:
This function is often used to clear a memory range, so it makes sens t o allow for `NULL` `image_filename` for that purpose.

Reply:
Makes sense, will add this to its documentation.